pgx/src/datum/range.rs

```diff
@@ -78,8 +78,9 @@ where
             None
         } else {
             let ptr = datum.cast_mut_ptr();
+            // Datum should be non-null and point to PG RangeType
             let range_type =
-                pg_sys::pg_detoast_datum(datum.cast_mut_ptr()) as *mut pg_sys::RangeType;
+                unsafe { pg_sys::pg_detoast_datum(datum.cast_mut_ptr()) as *mut pg_sys::RangeType };
```

Comment:
AHA! I knew there was going to be one of these hiding. Thank you for taking care of this.